packages/coin98-wallet/src/lib/coin98-wallet.ts

```diff
@@ -0,0 +1,167 @@
+import { isMobile } from "is-mobile";
+import type {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  InjectedWallet,
+  Account,
+  Optional,
+  Transaction,
+} from "@near-wallet-selector/core";
+import { getActiveAccount } from "@near-wallet-selector/core";
+import { waitFor } from "@near-wallet-selector/core";
+import type { InjectedCoin98 } from "./injected-coin98-wallet";
+import { signTransactions } from "@near-wallet-selector/wallet-utils";
+import type { FinalExecutionOutcome } from "near-api-js/lib/providers";
+
+declare global {
+  interface Window {
+    coin98: InjectedCoin98;
+  }
+}
+
+export interface Coin98WalletParams {
+  iconUrl?: string;
+}
+
+interface Coin98WalletState {
+  wallet: InjectedCoin98;
+}
+
+const isInstalled = () => {
+  return waitFor(() => !!window.coin98).catch(() => false);
+};
+
+const setupCoin98WalletState = (): Coin98WalletState => {
+  const wallet = window.coin98!;
+
+  return {
+    wallet,
+  };
+};
+
+const Coin98Wallet: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  store,
+  provider,
+  logger,
+}) => {
+  const _state = setupCoin98WalletState();
+
+  const getAccounts = (): Array<Account> => {
+    const accountId = _state.wallet.near.account;
+
+    if (!accountId) {
+      return [];
+    }
+
+    return [{ accountId: _state.wallet.near.account }];
+  };
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId" | "receiverId">>
+  ): Array<Transaction> => {
+    const { contract } = store.getState();
+
+    if (!contract) {
+      throw new Error("Wallet not signed in");
+    }
+
+    const account = getActiveAccount(store.getState());
+
+    if (!account) {
+      throw new Error("No active account");
+    }
+
+    return transactions.map((transaction) => {
+      return {
+        signerId: transaction.signerId || account.accountId,
+        receiverId: transaction.receiverId || contract.contractId,
+        actions: transaction.actions,
+      };
+    });
+  };
+
+  return {
+    async signIn({ contractId }) {
+      const existingAccounts = getAccounts();
+
+      if (existingAccounts.length) {
+        return existingAccounts;
+      }
+
+      await _state.wallet.near.connect({ prefix: "near_selector", contractId });
+      return getAccounts();
+    },
+
+    async signOut() {
+      // Ignore if unsuccessful (returns false).
+      await _state.wallet.near.disconnect();
+    },
+
+    async getAccounts() {
+      return getAccounts();
+    },
+
```

Comment:
The `BaseWalletBehaviour` has a new `verifyOwner(params: VerifyOwnerParams): Promise<VerifiedOwner | void>;` function so you have to add it in this return object. You can check out for example the `math-wallet` package and copy how they did it.

Reply:
commented fixed